gohcl/decode_test.go

```diff
@@ -643,3 +849,9 @@ func (e *fixedExpression) StartRange() (r hcl.Range) {
 func (e *fixedExpression) Variables() []hcl.Traversal {
 	return nil
 }
+
+func makeInstantiateType(target interface{}) func() interface{} {
+	return func() interface{} {
+		return reflect.New(reflect.TypeOf(target))
```

Comment:
...and to make this one compatible with all of the hand-written ones, we can make this unwrap the `reflect.Value` to return:

```suggestion
		return reflect.New(reflect.TypeOf(target)).Interface()
```